plugins/hydra_optuna_sweeper/hydra_plugins/hydra_optuna_sweeper/_impl.py

```diff
@@ -200,18 +214,58 @@ def sweep(self, arguments: List[str]) -> None:
             returns = self.launcher.launch(overrides, initial_job_idx=self.job_idx)
             self.job_idx += len(returns)
             for trial, ret in zip(trials, returns):
-                study._tell(trial, optuna.trial.TrialState.COMPLETE, [ret.return_value])
+                values: Optional[List[float]] = None
+                state: optuna.trial.TrialState = optuna.trial.TrialState.COMPLETE
+                func_error: Optional[Exception] = None
+                try:
+                    if len(directions) == 1:
+                        values = [float(ret.return_value)]
+                    else:
+                        if not isinstance(ret.return_value, (list, tuple)):
+                            raise ValueError(
+                                f"Return value must be a float value. Got {ret.return_value}."
+                            )
+                        values = [float(v) for v in ret.return_value]
+                        if len(values) != len(directions):
+                            raise ValueError(
+                                "The number of the values and the number of the objectives are"
+                                f" mismatched. Expect {len(directions)}, but actually "
+                                f"{len(values)}."
+                            )
+                except Exception as e:
+                    state = optuna.trial.TrialState.FAIL
+                    func_error = e
+                finally:
+                    study._tell(trial, state, values)
```

Comment:
That's a bit creative usage of finally.
I think it's better to do:
```python
try:
  ...
  study._tell(trial, state, values)
except Exception as e:
  study._tell(trial, optuna.trial.TrialState.FAIL, values)
  raise e
  
```

Reply:
That makes sense. I fixed it in https://github.com/facebookresearch/hydra/pull/1294/commits/22e4e528af58f6321b4c03814fc5e95974cae4ec.